File: app/src/main/java/com/examle/curexchange/ui/base/BasePresenter.java
package com.examle.curexchange.ui.base;

import com.arellomobile.mvp.MvpPresenter;

public abstract class BasePresenter<T extends BaseView> extends MvpPresenter<T> {

    void showError(String error) {
        getViewState().showToast(error);
    }

    public void loadProgress(boolean isProgress){
        getViewState().showProgress(isProgress);
    }
}

File: app/src/main/java/com/examle/curexchange/ui/base/BasePresenter.kt
package com.examle.curexchange.ui.base

import com.arellomobile.mvp.MvpPresenter

abstract class BasePresenter<T : BaseView> : MvpPresenter<T>() {

    fun showError(error: String) {
        viewState.showToast(error)
    }

    fun loadProgress(isProgress: Boolean){
        viewState.showProgress(isProgress)
    }
}